gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -140,23 +141,41 @@ private void writePolicyMetadata(final Map<String, PolicyMetadata> policyMetadat
     }
 
     /**
-     * This method finds policy dependencies from the bundle dependency graphf for a given policy id
+     * This method finds the policy dependencies from the bundle dependency graph for a given policy id
      * @param id String
      * @param rawBundle Bundle
      * @return Set
      */
-    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle) {
+    private Set<Dependency> getPolicyDependencies(final String id, final Bundle rawBundle){
+        final Set<Dependency> dependencyList = new HashSet<>();
         Map<Dependency, List<Dependency>> dependencyListMap = rawBundle.getDependencyMap();
         if (dependencyListMap != null) {
-            Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
-            for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
-                Dependency parent = entry.getKey();
-                if (parent.getId().equals(id)) { // Search for "id" to get its dependencies
-                    return new HashSet<>(entry.getValue());
+            populateDependencies(dependencyListMap, id, id, dependencyList);
+        }
+        return dependencyList;
+    }
+
+    /**
+     * This method recursively loads the entity dependencies from the given entity id
+     *
+     * @param dependencyListMap List
+     * @param rootId String
+     * @param id String
+     * @param dependencies Set
+     */
+    private void populateDependencies(Map<Dependency, List<Dependency>> dependencyListMap, String rootId, String id, Set<Dependency> dependencies) {
+        Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
+        for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
+            Dependency parent = entry.getKey();
+            if (parent.getId().equals(id)) {
+                List<Dependency> dependencyList = entry.getValue();
+                for (Dependency dependency : dependencyList) {
+                    if (!rootId.equals(dependency.getId()) && dependencies.add(dependency) && !BuilderConstants.NON_ENV_ENTITY_TYPES.contains(dependency.getType())) {
+                        populateDependencies(dependencyListMap, rootId, dependency.getId(), dependencies);
+                    }
```

Comment:
You can make this logic simple to read by moving out the recursive call for Env entities outside of this method. Like,

```
for (Dependency dependency : dependencyList) {
    if (!NON_ENV_ENTITY_TYPES.contains(dependency.getType())) {
        populateEnvTransitiveDependencies();
    } else {
        dependencies.add(dependency);
    }
}

private void populateEnvTransitiveDependencies() { ... }
```

Reply:
@nikhilahuja , why do we need separate method, that looks similar to existing method.
If it is just only for readability we can add proper comment over it.